### PR TITLE
Inject Guzzle Client into Transport

### PR DESCRIPTION
**Pull request by MikaFima**

### CODE REVIEW

File: Resources/config/services/transport.xml [hunk 1]
```diff
@@ -9,7 +9,9 @@
     </parameters>
 
     <services>
-        <service id="rezzza.process_one.transport.guzzle" class="%rezzza.process_one.transport.guzzle.class%" />
+        <service id="rezzza.process_one.transport.guzzle" class="%rezzza.process_one.transport.guzzle.class%">
+            <argument type="service" id="device.http.client"/>
```
Comment (stephpy) [inline Resources/config/services/transport.xml]: Where this client is defined ?

**Comment by MikaFima** [comment]
@tyx [IronArm] This one allow to set your own guzzle client, really not bc break

File: Transport/GuzzleTransport.php [hunk 2]
```diff
@@ -5,22 +5,32 @@
...
+    }
+
+    /**
      * {@inheritdoc}
      */
     public function send($url, $payload, array $parameters)
     {
-        $client = new Client();
+        $client = is_null($this->client) === false ? $this->client : new Client();
         $client->setSslVerification(false, true, 2);
```
Comment (MikaFima) [inline Transport/GuzzleTransport.php]: Maybe I have to clone client in order to keep source client intact ?

**Comment by tyx** [comment]
Can we directly inject Guzzle Client through the construct ?

We will tag a new version if needed, no problem.

**Comment by shouze** [comment]
Closing it, will merge https://github.com/rezzza/ProcessOneBundle/pull/9 instead.

**Pull request closed without merging**